tests/copilot_helper/utils/fixtures/test_service_manifest_with_copilot_bootstrap_image.yml

```diff
@@ -0,0 +1,112 @@
+# Generated by copilot-helper v0.1-TEST / 2023-08-22 16:00:00
+# The manifest for the "test-service" service.
+# Read the full specification for the "Load Balanced Web Service" type at:
+#  https://aws.github.io/copilot-cli/docs/manifest/lb-web-service/
+
+# Your service name will be used in naming your resources like log groups, ECS services, etc.
+name: test-service
+type: Load Balanced Web Service
+
+# Distribute traffic to your service.
+http:
+  # Requests to this path will be forwarded to your service.
+  # To match all requests you can use the "/" path.
+  path: '/'
+  # You can specify a custom health check path. The default is "/".
+  # healthcheck: '/'
+  target_container: nginx
+  healthcheck:
+    path: '/'
+    port: 8080
+    success_codes: '200,301,302'
+    healthy_threshold: 3
+    unhealthy_threshold: 2
+    grace_period: 180s
+
+sidecars:
+  nginx:
+    port: 443
+    image: public.ecr.aws/uktrade/nginx-reverse-proxy:latest
+    variables:
+      SERVER: localhost:8000
+
+  ipfilter:
+    port: 8000
+    image: public.ecr.aws/uktrade/ip-filter:latest
+    variables:
+      PORT: 8000
+      SERVER: localhost:8080
+      APPCONFIG_PROFILES: ipfilter:default:default
+      IPFILTER_ENABLED: True
+      EMAIL: sre@digital.trade.gov.uk
+
+  appconfig:
+    port: 2772
+    image: public.ecr.aws/aws-appconfig/aws-appconfig-agent:2.x
+    essential: true
+    variables:
+      ROLE_ARN: arn:aws:iam::763451185160:role/AppConfigIpFilterRole
+
+# Configuration for your containers and service.
+image:
+  location: public.ecr.aws/uktrade/copilot-bootstrap:latest
```

Comment:
A whole new file for only one line difference 😬 

I don't like it 🤮 

I'm wondering whether it would be better to just grab `test_service_manifest.yml`s contents and change this one line before writing the `manifest.yml` used in the test.

The same could be said for `test_service_manifest_with_mismatched_service.yml`.

If we think my suggestion is worthy, I feel it should be done under its own ticket as there are undoubtedly loads of other places we could apply the same approach to reduce the amount of files we need to maintain.

Reply:
Another option is to use a similar approach to [ci-image-builder](https://github.com/uktrade/ci-image-builder/blob/main/test/image_builder/codebase/test_codebase.py#L29) and have functions that can write to the fakefs a parameterised version of the file. This would also have the benefit of reducing file IO which is one of the reasons the test suite takes a full 30 seconds to run...

Reply:
https://uktrade.atlassian.net/browse/DBTP-552

The approach @antroy-madetech and I took in [the pipeline command tests](https://github.com/uktrade/copilot-tools/blob/main/tests/copilot_helper/test_command_pipeline.py#L133-L141) to stop the tests being so brittle about whitespace chnages, or adding todo comments etc. my be relevant too.